guides/keras_cv/object_detection_keras_cv.py

```diff
@@ -0,0 +1,848 @@
+
+Next let's load an image:
+"""
+
+filepath = tf.keras.utils.get_file(origin="https://i.imgur.com/gCNcJJI.jpg")
+image = keras.utils.load_img(filepath)
+image = np.array(image)
+
+visualization.plot_image_gallery(
+    [image],
+    value_range=(0, 255),
+    rows=1,
+    cols=1,
+    scale=5,
+)
+
+"""
+To use the `RetinaNet` architecture with a ResNet50 backbone, you'll need to
+resize your image to a size that is divisible by 64.  This is to ensure
+compatibility with the number of downscaling operations done by the convolution
+layers in the ResNet.
+
+If the resize operation distorts
+the input's aspect ratio, the model will perform signficantly poorer.  For the
+pretrained `"retinanet_resnet50_pascalvoc"` preset we are using, the final
+`MeanAveragePrecision` on the `pascalvoc/2012` evaluation set drops to `0.15`
+from `0.38` when using a naive resizing operation.
+
+Additionally, if you crop to preserve the aspect ratio as you do in classification
+your model may entirely miss some bounding boxes.  As such, when running inference
+on an object detection model we recommend the use of padding to the desired size,
+while resizing the longest size to match the aspect ratio.
+
+KerasCV makes resizing properly easy; simply pass `pad_to_aspect_ratio=True` to
+a `keras_cv.layers.Resizing` layer.
+
+This can be implemented in one line of code:
+"""
+
+inference_resizing = keras_cv.layers.Resizing(
+    640, 640, pad_to_aspect_ratio=True, bounding_box_format="xywh"
+)
+
+"""
+This can be used as our inference preprocessing pipeline:
+"""
+
+image_batch = inference_resizing([image])
+
+"""
+`keras_cv.visualization.plot_bounding_box_gallery()` supports a `class_mapping`
+parameter to highlight what class each box was assigned to.  Let's assemble a
+class mapping now.
+"""
+
+class_ids = [
+    "Aeroplane",
+    "Bicycle",
+    "Bird",
+    "Boat",
+    "Bottle",
+    "Bus",
+    "Car",
+    "Cat",
+    "Chair",
+    "Cow",
+    "Dining Table",
+    "Dog",
+    "Horse",
+    "Motorbike",
+    "Person",
+    "Potted Plant",
+    "Sheep",
+    "Sofa",
+    "Train",
+    "Tvmonitor",
+    "Total",
+]
+class_mapping = dict(zip(range(len(class_ids)), class_ids))
+
+"""
+Just like any other `keras.Model` you can predict bounding boxes using the
+`model.predict()` API.
+"""
+
+y_pred = pretrained_model.predict(image_batch)
+# y_pred is a bounding box Tensor:
+# {"classes": ..., boxes": ...}
+visualization.plot_bounding_box_gallery(
+    image_batch,
+    value_range=(0, 255),
+    rows=1,
+    cols=1,
+    y_pred=y_pred,
+    scale=5,
+    font_scale=0.7,
+    bounding_box_format="xywh",
+    class_mapping=class_mapping,
+)
+
+"""
+In order to support easy this easy and intuitive inference workflow, KerasCV
+performs non-max suppression inside of the `RetinaNet` class.
+Non-max suppression is a traditional computing algorithm that solves the problem
+of a model detecting multiple boxes for the same object.
+
+**Example of pre-NMS detections:**
```

Comment:
Couldn't this come from the API itself? We just have to overwrite the prediction decoder twice, right?

Reply:
it could, but I feel its weird to write to it for that demonstration purpose.  I wrote it out and it just made the guide look a bit strange.   Also, this isnt a truly non-nmsed image, its just a "barely nmsed" image.  Feel its burdening the user with maybe more logic than they need

Reply:
actually, this is better !  Updated